### ISSUE TITLE
Add vertical slider to "video comparison" tool

### ISSUE THREAD

**Issue by ukmark62** [staxrip/staxrip#702]
I would like to request a vertical slider be added to the "video comparison" tool. This slider would be a vertical line from the top of the panel to the bottom and would default to be the middle of the screen. 

Both comparison videos would be loaded (one video on top of the other) and initially, the left half the video would be a frame from video A and the other half wiould be the same frame from video B. Dragging the screen to the left or right would show more of one of the video frames.  I.E. dragging slider to the left would show more of video B and dragging it to the right would show more of video B. This way you can see the same frame from both videos at the same time on a single screen.

There are no alternatives that I am aware of.

The feature would make it easier to directly compare the output from 2 identical videos that may have been encoded in different ways, or to show an original vs encoded video comparison.

There is an HTML5 example here. I realise this is not HTML5 but this link shows the principle of it. The gif at the top of this link shows the comparison slider in action, I have also pasted the gif file below the link:-
https://codemyui.com/video-comparison-slider-animation-using-html5/

![image](https://user-images.githubusercontent.com/18571394/117585812-5ba20200-b10c-11eb-8bf8-2314db1d5365.png)

Thanks.

**Comment by JJKylee** [comment]
Looks fancy but IMHO, implementing this feature in VB may be very challenging.
But I can be wrong. Let's see if @Dendraspis or @stax76 (or any other willing to participate) is intrigued by this. 🤔

**Comment by ukmark62** [comment]
I agree it may be complicated and to be fair, the existing video comparison tool works well, but it would be a nice to have.

**Comment by ukmark62** [comment]
Yes, agreed - a new tool to allow side-by-side comparison of 2 videos.

**Comment by Patman86** [comment]
The player could serve as a basis, right?
[video-compare](https://github.com/pixop/video-compare/tree/20200724)

**Comment by ukmark62** [comment]
Absolutely – great find! Sent from Mail for Windows 10 From: Patman86Sent: 22 May 2021 1:58 PMTo: staxrip/staxripCc: Mark; AuthorSubject: Re: [staxrip/staxrip] Add NEW tool to allow side-by-side comparison of 2 videos (#702) The player could serve as a basis, right?video-compare—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by JJKylee** [comment]
> The player could serve as a basis, right?
> [video-compare](https://github.com/pixop/video-compare/tree/20200724)

FYI, I just made a very simple GUI for `video-compare` in Python/tkinter, and it's a single executable file.
[Here](https://github.com/JJKylee/video-compare_GUI/releases).
It's totally free, so do whatever you want to do. 😄

**Comment by ukmark62** [comment]
Thanks, just tried it and it works great. Appreciate your efforts!!

**Comment by JJKylee** [comment]
There's a caveat:

If you add `video-compare_GUI.exe` as a StaxRip menu item (`ExecuteCommandLine` item), the window called by `video-compare.exe` fails to launch from within `video-compare_GUI.exe`.

I guess it's because the Python `subprocess` of `video-compare.exe` called by `video-compare_GUI.exe` conflicts with the native StaxRip process call module. I have no idea how to avoid this as of now. I need to investigate.

So, if we want to call `video-compare.exe` _from within StaxRip_, we need to _compose a separate GUI dialog in StaxRip_ that calls `video-compare.exe` a la `mtn.exe` in the past. Maybe @Dendraspis or @stax76 can do this, but I'm not sure if he/they are willing. ...Don't hate me, guys. 😅

**Comment by ukmark62** [comment]
Just the fact that you have created a working front-end GUI is good enough for myself. It would be great to incorporate into StaxRip, but I am grateful for all your efforts. Sent from Mail for Windows 10 From: JKyleSent: 03 June 2021 7:45 PMTo: staxrip/staxripCc: Mark; AuthorSubject: Re: [staxrip/staxrip] Add NEW tool to allow side-by-side comparison of 2 videos (#702) There's a caveat:If you add video-compare_GUI.exe as a StaxRip menu item (ExecuteCommandLine item), the window called by video-compare.exe fails to launch from within video-compare_GUI.exe.I guess it's because the Python subprocess of video-compare.exe called by video-compare_GUI.exe conflicts with the native StaxRip process call module. I have no idea how to avoid this as of now. I need to investigate.So, if we want to call video-compare.exe from within StaxRip, we need to compose a separate GUI dialog in StaxRip that calls video-compare.exe a la mtn.exe in the past. Maybe @Dendraspis or @stax76 can do this, but I'm not sure if he/they are willing. ...Don't hate me, guys. 😅—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by JJKylee** [comment]
Wow... _Windows 10 Mail_ app surely seems to do something on the format of the reply post. 😓

Anyway... @Dendraspis, since @ukmark62 seems satisfied with my 'solution', I think you can close this thread _unless_ you're interested in embedding `video-compare` in StaxRip. 😁

BTW, @Patman86, can you update and build `video-compare` on your MSYS2? If you're interested, can you also change the hard-wired keyboard shortcuts a la StaxRip preview style? Currently, `video-compare` up/down arrows and pageup/pagedown arrows move the video in the opposite direction to StaxRip's.

And I've found a bug with `video-compare` already: if the window is zoomed, the mouse cursor location is misaligned with the vertical slider. I think I need to report this on its repo. 🤔

**Comment by Dendraspis** [comment]
I leave it open for now. There are some other things to do first. 😉
I'm also not sure what to do with it ... we will see. 😁
